front/src/pages/ExpensesPage/index.tsx

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable react/no-unescaped-entities */
+import {
+  Box,
+  ButtonGroup,
+  IconButton,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Theme,
+  Typography,
+  useMediaQuery,
+} from "@mui/material";
+import variables from "../../variables";
+import { useLazyQuery, useMutation } from "@apollo/client";
+import {
+  DELETE_EXPENSE_BY_ID,
+  GET_EXPENSE_BY_USER_ID,
+} from "../../gql/ExpenseGql";
+import { useContext, useEffect, useState } from "react";
+import { ExpenseType } from "../../types/expense";
+import { Delete } from "@mui/icons-material";
+import AddExpenseModal from "../../components/AddExpenseModal/AddExpenseModal";
+import CategoryBar from "./utils/CategoryBar";
+import EditExpenseModal from "./utils/EditExpenseModal";
+import { LoginContext } from "../../hooks/useLoginContext";
+
+export default function ExpensesPage() {
+  const { userId } = useContext(LoginContext);
+  const isLg = useMediaQuery((theme: Theme) => theme.breakpoints.up("lg"));
+
+  const inputStyle = {
+    fontFamily: "Roboto",
+    fontSize: "1rem",
+    title: {
+      color: variables.thirdColor,
+      fontWeight: "bold",
+    },
+  };
+
+  const [fetchExpensesByUserId, { data: dataExpenses }] = useLazyQuery<{
+    getAllExpensesByUserId: ExpenseType[];
+  }>(GET_EXPENSE_BY_USER_ID);
```

Comment:
```suggestion
  const { data: dataExpenses, error, loading } = useQuery<{
    getAllExpensesByUserId: ExpenseType[];
  }>(GET_EXPENSE_BY_USER_ID);
  
  if(error) {
  return <UnComposantError /> ou alors une string
  }
  
  if (loading) {
  return <UnLoader /> ou alors une string
  }
```

Juste pour la remarque, on peut aussi gérer cette query de cette manière, ça permet de retirer le useEffect plus bas qui dépend de `fetchExpensesByUserId` mais attention il faut mettre les conditions du loading et error après les hooks sinon il y aura une erreur React qui dit qu'il n'y a pas le même nombre de hooks à chaque render.

En général, on utilisera `useLazyQuery` si on a besoin de lancer la query sur un event.